ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/AbstractMiningCoordinator.java

```diff
@@ -40,6 +44,8 @@
         M extends BlockMiner<? extends AbstractBlockCreator>>
     implements BlockAddedObserver, MiningCoordinator {
 
+  private final Map<String, SealerInfo> sealerInfos;
```

Comment:
I think a guava cache may be more appropriate for this particular data set https://github.com/google/guava/wiki/CachesExplained - we can set the max size via the flag and we can expire values after a certain time.  Which given miners come and go I think is a good thing.  Perhaps 10 minutes?  Also set by a flag like the max cache size.

Reply:
Done. The new policy is to delete after 10 minutes of inactivity. And the limit is by default 1000 but configurable via the CLI

Reply:
I forgot I just need to add the CLI flag for TTL

Reply:
Done you need to use `--Xminer-remote-sealers-hashrate-ttl=10`(default 10 minutes)